src/connectors/tinyman/tinyman.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  poolUtils,
+  SignerTransaction,
+  SupportedNetwork,
+  Swap,
+  SwapQuote,
+  SwapQuoteType,
+  SwapType,
+  V2PoolInfo,
+} from '@tinymanorg/tinyman-js-sdk';
+import { V2SwapExecution } from '@tinymanorg/tinyman-js-sdk/dist/swap/types';
+import { Account } from 'algosdk';
+import { pow } from 'mathjs';
+import { PriceRequest } from '../../amm/amm.requests';
+import { Algorand } from '../../chains/algorand/algorand';
+import { AlgorandAsset } from '../../chains/algorand/algorand.requests';
+import { percentRegexp } from '../../services/config-manager-v2';
+import {
+  HttpException,
+  TOKEN_NOT_SUPPORTED_ERROR_CODE,
+  TOKEN_NOT_SUPPORTED_ERROR_MESSAGE,
+} from '../../services/error-handler';
+import { logger } from '../../services/logger';
+import { TinymanConfig } from './tinyman.config';
+
+export class Tinyman {
+  private static _instances: { [name: string]: Tinyman };
```

Comment:
Let's use `LRUCache` here to avoid any potential for memory leaks — code can be directly adapted from `Algorand`'s

Reply:
done